test/distributed/fsdp/test_fsdp_comm.py

```diff
@@ -149,9 +149,9 @@ def _get_ref_num_all_gathers_in_pass(
                 f"sharding_strategy={sharding_strategy}"
         if is_first_iter and pass_type == PassType.FWD:
             # With execution order validation, on the first iteration, we have
-            # an additional all-gather before every actual all-gather in the
-            # forward pass
-            num_all_gathers *= 2
+            # an additional two all-gathers before every actual all-gather in
```

Comment:
The first all-gather communicates how many handles the `forward()` plans to all-gather. The second all-gather communicates the indices representing each handle. We need the first all-gather in the general case because otherwise we do not know how many indices to communicate in the second all-gather.